### ISSUE TITLE
[FEATURE REQUEST] Transparent References/Pointers

### ISSUE THREAD

**Issue by jharmer95** [jharmer95/rpc.hpp#39]
# Feature Request: Transparent References/Pointers

## Brief Summary of Request

Currently, a call to a function with references or pointers requires re-binding the value after the call. It would instead be preferable to have the values already assigned at the end of the call, either (ideally) by simply using the actual reference or pointer directly on the client-side, or (if the first option is not feasible) by moving the value to the pointer/reference.

## What Is The Value Added With This Feature

- More elegant API
  - Basically equivalent to an actual function call
- Potentially slightly faster/use less memory for copy of pointed/referenced object(s)

## Desired Outcome

```C++
// server
void AddOneToNum(int* num)
{
    *num += 1;
}

// client
int n = 0;
rpc::call<Serial>(client, "AddOneToNum", &n);
assert(n == 1);
```
